### ISSUE TITLE
Notes about initial setup and comparison to macOS

### ISSUE THREAD

**Issue by geerlingguy** [geerlingguy/pi-dev-playbook#3]
Following are some of the settings I've changed just for appearance and initial usage:

  - Removed desktop picture.
  - Set desktop background to medium grey (solid color).
  - Set Dock/Pi bar to bottom, interface defaults to 'Large'.
  - Mouse: Acceleration 7.0, Double-click delay 250
  - Keyboard: Repeat delay: 500, Repeat interval: 30

Regarding 4K, the Pi is capable of 4K resolution, but with a number of caveats:

  1. You can do up to 30 Hz, so there is noticeable 'flicker' while scrolling, moving windows, etc. (jarring to my eyes).
  2. The default window manager/UI does not natively have any kind of HiDPI mode built in.
  3. You can set many apps to higher font sizes but many apps end up looking a little janky.
  4. You can enable 'pixel doubling', but this basically doubles pixels from 1080p resolution, though without anti-aliasing, so (to my eye, at least) it is harder to read (and sticks with 30 Hz refresh rate), so just using 1080p is a better option.

Notes on the keyboard and mouse:

  - As with any time I switch platforms (Mac to Windows and vice-versa, especially), my muscle memory for control/option/command/alt really hinders my ability to be productive the first few hours, because I keep accidentally closing something or going back when trying to paste.
  - The way mouse acceleration works makes my Apple Magic Trackpad almost completely unusable. I'll have to see if there's a better driver for it I could use. It doesn't feel natural at all. Luckily scroll behavior is pretty good still. Also, can't right click without a driver for it. (For now I'm using my Kensington USB mouse, but boy do I hate mice!).
  - The keyboard responsiveness is great, no problems there.

**Comment by geerlingguy** [comment]
The way the bash profile is set up is a little confusing with the default Terminal app (which is a bit lacking in some areas, but is generally adequate. I had to manually add the following bit to `~/.bashrc` because `~/.profile` is not loaded by the Terminal, nor could I figure out a way to get it to load (there is no ~/.bash_profile` in my home dir):

```
# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/.local/bin" ] ; then
    PATH="$HOME/.local/bin:$PATH"
fi
```

Without that block, installs inside `~/.local/bin` are not found (e.g. Ansible installed via `pip3`).

**Comment by geerlingguy** [comment]
While setting up my iCloud email account through Evolution, I had to [Set an app-specific password](https://support.apple.com/en-us/HT204397), otherwise Evolution just kept prompting me for my password over and over again.

**Comment by geerlingguy** [comment]
How the heck do you type an em-dash in Linux??

> Hold down one of the Alt keys and type on the numeric keypad: 0150 for an en dash or 0151 for an em dash.

That doesn't work. I'm so used to special characters being something like Shift + option + dash on the Mac, this 'hold a key and type a character code' voodoo doesn't make any sense. No wonder nobody knows the difference between a dash, en dash, or em dash!

**Comment by geerlingguy** [comment]
In terms of speed, most things are fine so far. Slack in Chromium is tolerable after it loads, but first load/paint takes like 5-10 seconds.

**Comment by geerlingguy** [comment]
I also futzed around with the system appearance and attempted to automate the selection of a 'darker' theme (though it's not Dark/Night mode) in GTK 3.0 but (a) I can't make it be a true dark/night mode, and that is not something that trickles down, e.g. to Chromium's `prefers-dark-mode` media query, and (b) I tried many methods of automating the switch, but in the end I still have to run `lxappearance` to switch themes.

Ugh... and yes, I know "I could just choose one of 20 other GUI interfaces for Linux", but that's not a fun thing to do for someone who just wants to have a not-glaringly-bright interface all the time :P

**Comment by geerlingguy** [comment]
I was testing out Zoom, Slack, and Google Hangouts in Chromium (since none of the native apps for the first two are available on ARM64, though they do have Linux AMD/X86 builds). Try as I might, I could not get any of the apps to recognize audio either through my attached Logitech C920 webcam, nor through my Behringer U-Phoria UMC202HD audio interface (both plugged in through a powered Amazon Basics USB 3.0 powered hub, which has not exhibited problems with either my MacBook Pro or my Dell XPS 13 in the past).

Bluejeans at least _tried_ to let me select a mic, though it failed after a couple seconds saying I needed to allow it access via the browser (I had, the first time, and I confirmed in Chromium's settings that Bluejeans had access to the camera and microphone, and I even tried setting the default microphone input in Chromium's settings). Zoom wouldn't even let me attempt to enable computer audio. And Bluejeans also bailed on my when I tried doing phone audio.

All three services seemed to be fine with my webcam for video, at least, though Zoom recognized it the first try, then failed from there on out.

In other words... A/V in Linux is still annoyingly difficult years later (I remember having similar flaky issues on a super old Red Hat box back in the day). On my XPS 13 running Fedora, it seems the drivers for at least the built-in devices were good enough to work consistently.

Anyways, I'm going to see if I can do some Audio/Video recording with my mic and webcam, at least.

**Comment by geerlingguy** [comment]
For audio, I can get my audio device using:

```
$ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: b1 [bcm2835 HDMI 1], device 0: bcm2835 HDMI 1 [bcm2835 HDMI 1]
  Subdevices: 4/4
  Subdevice #0: subdevice #0
  Subdevice #1: subdevice #1
  Subdevice #2: subdevice #2
  Subdevice #3: subdevice #3
card 1: Headphones [bcm2835 Headphones], device 0: bcm2835 Headphones [bcm2835 Headphones]
  Subdevices: 4/4
  Subdevice #0: subdevice #0
  Subdevice #1: subdevice #1
  Subdevice #2: subdevice #2
  Subdevice #3: subdevice #3
card 2: U192k [UMC202HD 192k], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

That seems to correspond to `-i hw:2,0`, but I could be wrong...

**Comment by geerlingguy** [comment]
[This thread](https://www.raspberrypi.org/forums/viewtopic.php?t=206692) helped me figure out the audio issue; it has to do with `alsa` not being able to load the device correctly, it seems. I had to create an `.asoundrc` file with my preferences:

```
$ cat ~/.asoundrc
pcm.!default {
        type hw
        card 2
}

ctl.!default {
        type hw
        card 2
}
```

See docs: https://www.alsa-project.org/main/index.php/Asoundrc

But I was still getting the following error:

```
[alsa @ 0x55b31e38c0] cannot set sample format 0x10000 2 (Invalid argument)
hw:2,0: Input/output error
```

**Comment by geerlingguy** [comment]
Video discussion moved to https://github.com/geerlingguy/pi-dev-playbook/issues/4

**Comment by geerlingguy** [comment]
All right. One day in... I can do some of my open source dev work, but not much of my other productivity work, at least not with tools that make it quick and easy.

I think I'm going to have to call it a day on this project. I was planning on going a few days, but there are enough rough edges in _my_ desktop workflow that I could not live with the limitations and constant pressure of having some apps work sometimes and not others.

Now: for a person who really only needs an editor, terminal, browser, and Docker? This is a totally workable workstation. If you don't need to do multimedia work, and if you don't rely on some apps which are strongly tied to macOS or Windows and have a fancy/nice GUI, then you could be very productive with a Raspberry Pi setup that costs around $150 or less.

But you have to be willing to go through a pretty severe learning curve. For kids and beginners, I would recommend a Pi for learning purposes, but not as a standard "pick this up and start being creative / have fun" type of tool. Unless you're like me. I would've been totally into a Pi when I was a kid. But I'm the type of kid who spent New Years Eve getting DOS upgraded on a hacked-together 386 so I could play Doom one year...
